sdk/communication/AzureCommunication/Tests/ObjCCommunicationTokenCredentialAsyncTests.m

```diff
@@ -73,18 +74,26 @@ - (void)xtest_ObjCRefreshTokenProactivelyTokenExpiringInOneMin {
         XCTAssertEqual(accessToken.token, weakSelf.sampleToken);
         XCTAssertEqual(weakSelf.fetchTokenCallCount, 1);
         
-        [expectation fulfill];
```

Comment:
if we don't call expection fulfill, are we removing expectation? I don't see the removal above line 76.

Reply:
Yes I changed up the implementation to not rely on `expectation`. Using some advice from Java and .net platform. I tried manipulating the run loop up until  the `loopUntil` similar to the custom clock in .net.

Reply:
should there be code to remove the declaration of 'expectation'? something like this 
![image](https://user-images.githubusercontent.com/8701731/107580614-ac589c00-6bab-11eb-8699-0e4b47b44e8c.png)